Library/Homebrew/resource.rb

```diff
@@ -266,15 +266,15 @@ def stage_resource(prefix, debug_symbols: false, &block)
   private
 
   def detect_version(val)
-    return Version::NULL if val.nil? && url.nil?
-
-    case val
-    when nil     then Version.detect(url, **specs)
+    version = case val
+    when nil     then url.nil? ? Version::NULL : Version.detect(url, **specs)
     when String  then Version.create(val)
     when Version then val
     else
       raise TypeError, "version '#{val.inspect}' should be a string"
     end
+
+    version.null? ? nil : version
```

Comment:
```suggestion

    version unless version.null?
```

or, if the second suggestion above works:
```suggestion

    version
```

or could remove the `version` assignment altogether